### ISSUE TITLE
False-positive error on multiple exclusion occurrence in a form

### ISSUE THREAD

**Issue by mikebohdan** [clj-kondo/clj-kondo#1187]
**version**

v2021.02.28

**platform**

JVM

**problem**

In the recent version if the unresolved symbol exclusion appears in the form multiple times it will be a false-positive on the second occurrence.

**repro**

```
(ns app.core
  (:require [cheshire.core :as json]
            [slingshot.slingshot :refer [try+]])
  (:import [com.fasterxml.jackson.core JsonParseException]))

(defn get-message [raw-data]
  (try+
    (json/parse-string raw-data)
    (catch JsonParseException _
      (println {:msg "JSON parsing exception." :error (:throwable &throw-context)})
      raw-data)
    (catch Object _
      (println {:msg "Unexpected exception" :error (:throwable &throw-context)})
      raw-data)))
```
**config**

```

{:linters       {:not-empty?        {:level :off}
                 :unresolved-symbol {:exclude [(slingshot.slingshot/try+ [&throw-context e _])
                                               (clojure.core.match/match)
                                               (clojure.test.check.clojure-test/defspec)]}}
 :skip-comments true
 :output        {:pattern       "{{LEVEL}} {{filename}}:{{row}}:{{col}} - {{message}}"
                 :format        :text
                 :exclude-files []}}
```

**expected behavior**
Using version `v2020.11.07` (via brew) or `v2021.02.13` (via lein) all works fine and there are no errors found. Using the latest version `v2021.02.28` I have an error `ERROR src/app/core.clj:13:64 - Unresolved symbol: &throw-context` pointing to the second occurrence of the `&throw-exception` at the example above.

**Comment by borkdude** [comment]
Can you please include namespace forms to make this a standalone example?

**Comment by mikebohdan** [comment]
@borkdude, copied a whole file to the example.

**Comment by borkdude** [comment]
Minimal repro:

```
(ns app.core
  {:clj-kondo/config '{:linters {:unresolved-symbol {:exclude [(slingshot.slingshot/try+ [x])]}}}}
  (:require [slingshot.slingshot]))

(slingshot.slingshot/try+
 x
 (:foo x))
```

**Comment by borkdude** [comment]
@mikebohdan Workaround for now:

```
#_:clj-kondo/ignore
 (:foo x)
 ```

**Comment by borkdude** [comment]
@mikebohdan Some feedback on your config:

- `clojure.core.match` is now supported out of the box with clj-kondo, no need to exclude it anymore
- you can get better linting for `slingshot` with a hook: https://github.com/clj-kondo/config/tree/master/slingshot
